apps/prairielearn/elements/pl-excalidraw/pl-excalidraw.py

```diff
@@ -0,0 +1,100 @@
+import base64
+import json
+from pathlib import Path
+
+import chevron
+import lxml.html
+import prairielearn as pl
+
+
+def prepare(element_html: str, _: pl.QuestionData) -> None:
+    element = lxml.html.fragment_fromstring(element_html)
+    pl.check_attribs(element, ["name"], ["width", "height", "file", "file_dir"])
+
+
+def render(element_html: str, data: pl.QuestionData) -> str:
+    element = lxml.html.fragment_fromstring(element_html)
+    with open("pl-excalidraw.mustache", "r", encoding="utf-8") as template:
+        drawing_name = pl.get_string_attrib(element, "name")
+        initial_content: str = ""
+
+        def load_file_content() -> str:
+            file_dir = pl.get_string_attrib(
+                element, "file_dir", "client_files_question_path"
+            )
+            if file_dir not in data["options"]:
+                path_suffixes = [
+                    key for key in data["options"].keys() if key.endswith("_path")
+                ]
+                raise RuntimeError(f"{file_dir=} not a valid key ({path_suffixes})")
+            file = Path(data["options"][file_dir]) / pl.get_string_attrib(
+                element, "file"
+            )
+            if not file.exists():
+                raise RuntimeError(
+                    f"Drawing named {drawing_name} at {file} cannot be found"
+                )
+            return file.read_text(encoding="utf-8")
+
+        match data["panel"]:
+            case "answer":
+                # Answer must have a file attribute
+                if not pl.has_attrib(element, "file") and data["panel"] == "answer":
+                    raise RuntimeError(
+                        f"Answer drawing '{drawing_name}' does not have a `file` argument"
+                    )
+                initial_content = load_file_content()
+            case "question":
+                # First try loading the submission
+                if drawing_name in data["submitted_answers"]:
+                    initial_content = (
+                        data["submitted_answers"].get(drawing_name) or initial_content
+                    )
+                # Next, try using the file attribute to load the starter diagram
+                elif pl.has_attrib(element, "file"):
+                    initial_content = load_file_content()
+                # Finally, give up and mark it as empty
+                else:
+                    initial_content = ""
+            case "submission":
+                initial_content = (
+                    data["submitted_answers"].get(drawing_name) or initial_content
+                )
+            case panel:
+                raise RuntimeError(f"Unhandled panel type {panel}")
```

Comment:
This is what we do in other elements.

```suggestion
                assert_never(data["panel"])
```

Reply:
This _might_ require removing the match statement, but that should be done if this is the case, since the assert ensures that the cases are exhaustive. It might also require `assert_never(panel)`, since this is the local variable used in the match.

Reply:
Fair, I didn't actually check that this works, but the general idea of using `assert_never` is what I was going for.